### PR TITLE
Fix/#1602 - Wait with no timeout should wait indefinitely

### LINKED ISSUE #1602: [🐛 BUG] wait=True does not work for scenario submission
### What went wrong? 🤔

`wait=True` inside the scenario submission should wait for the submission to be finished. Here, it is still asynchronous and will not wait for the submission to be finished.

### Expected Behavior

This method with this parameter should wait for the submission to finish.

### Steps to Reproduce Issue

Run this code. We don't see any logs from the task when we should receive it.

```python
# Import necessary libraries
import pandas as pd
import taipy as tp
from taipy import Config, Scope, Frequency
import time

import datetime as dt

Config.configure_job_executions(mode="standalone", max_nb_of_workers=2)

# Function to run a Dataiku scenario
def run_something(input_1, input_2):
    datetime = dt.datetime.now()
    date = dt.date(2018, 1, 1)
    int_var = 10
    string_var = "String"
    for i in range(10):
        time.sleep(1)
        print("In loop in task", i)
    return datetime, date, int_var, string_var


data = {"toto": [i for i in range(10_000)], 
        "titi": [2*i for i in range(10_000)],
        "tata": [4*i for i in range(10_000)]}



input_1_cfg = Config.configure_data_node(
    id="input_1_data_node",
    default_data=data,
)


input_2_cfg = Config.configure_data_node(
    id="input_2_data_node",
    default_data=data,
)

datetime_cfg = Config.configure_data_node(id="datetime_data_node")
date_cfg = Config.configure_data_node(id="date_data_node")
int_cfg = Config.configure_data_node(id="int_data_node")
string_cfg = Config.configure_data_node(id="string_data_node")


# Scenario and task configuration in Taipy
scenario_task_cfg = Config.configure_task(
    id="scenario_task",
    function=run_something,
    input=[input_1_cfg, input_2_cfg],
    output=[datetime_cfg, date_cfg, int_cfg, string_cfg]
)

scenario_cfg = Config.configure_scenario(
    id="scenario",
    task_configs=[scenario_task_cfg],
    frequency=Frequency.DAILY)


# Main execution block with GUI setup
if __name__ == "__main__":
    tp.Core().run()
    scenario = tp.create_scenario(scenario_cfg)
    print("Before submit")
    scenario.submit(wait=True)
    print("After submit")
``` 
### Version of Taipy

develop - 7/17/24

### Acceptance Criteria

- [ ] Ensure new code is unit tested, and check code coverage is at least 90%.
- [ ] Create related issue in taipy-doc for documentation and Release Notes.

### Code of Conduct

- [X] I have checked the [existing issues](https://github.com/Avaiga/taipy/issues?q=is%3Aissue+).
- [ ] I am willing to work on this issue (optional)

### PR DESCRIPTION
**Pull request by trgiangdo**
Resolves #1602

### CODE REVIEW

File: taipy/core/_orchestrator/_orchestrator.py [hunk 1]
```diff
@@ -66,6 +66,7 @@ def submit(
                 finished in asynchronous mode.
              timeout (Union[float, int]): The optional maximum number of seconds to wait for the jobs to be finished
                 before returning.
```
Comment (FabienLelaquais) [inline taipy/core/_orchestrator/_orchestrator.py]:
```suggestion
                before returning.<br/>
```
Make sure the line break is also added in the other locations where the next line appears.

File: taipy/core/_orchestrator/_orchestrator.py [hunk 2]
```diff
@@ -66,6 +66,7 @@ def submit(
...
+                If not provided and wait is True, the function waits indefinitely.
```
Comment (FabienLelaquais) [inline taipy/core/_orchestrator/_orchestrator.py]:
```suggestion
                If not provided and *wait* is True, the function waits indefinitely.
```

File: taipy/core/_orchestrator/_orchestrator.py [hunk 3]
```diff
@@ -120,6 +121,7 @@ def submit_task(
                 in asynchronous mode.
              timeout (Union[float, int]): The optional maximum number of seconds to wait for the job
                 to be finished before returning.
+                If not provided and wait is True, the function waits indefinitely.
```
Comment (FabienLelaquais) [inline taipy/core/_orchestrator/_orchestrator.py]:
```suggestion
                If not provided and *wait* is True, the function waits indefinitely.
```

File: taipy/core/scenario/scenario.py [hunk 4]
```diff
@@ -609,6 +609,7 @@ def submit(
                 asynchronous mode.
             timeout (Union[float, int]): The optional maximum number of seconds to wait for the jobs to be finished
                 before returning.
+                If not provided and wait is True, the function waits indefinitely.
```
Comment (FabienLelaquais) [inline taipy/core/scenario/scenario.py]:
```suggestion
                If not provided and *wait* is True, the function waits indefinitely.
```

File: taipy/core/sequence/sequence.py [hunk 5]
```diff
@@ -312,6 +312,7 @@ def submit(
                 in asynchronous mode.
             timeout (Union[float, int]): The maximum number of seconds to wait for the jobs to be finished before
                 returning.
+                If not provided and wait is True, the function waits indefinitely.
```
Comment (FabienLelaquais) [inline taipy/core/sequence/sequence.py]:
```suggestion
                If not provided and *wait* is True, the function waits indefinitely.
```

File: taipy/core/taipy.py [hunk 6]
```diff
@@ -241,6 +241,7 @@ def submit(
             in asynchronous mode.
         timeout (Union[float, int]): The optional maximum number of seconds to wait
             for the jobs to be finished before returning.
+            If not provided and wait is True, the function waits indefinitely.
```
Comment (FabienLelaquais) [inline taipy/core/taipy.py]:
```suggestion
            If not provided and *wait* is True, the function waits indefinitely.
```

File: taipy/core/task/task.py [hunk 7]
```diff
@@ -232,6 +232,7 @@ def submit(
                 mode.
             timeout (Union[float, int]): The maximum number of seconds to wait for the job to be finished before
                 returning.
+                If not provided and wait is True, the function waits indefinitely.
```
Comment (FabienLelaquais) [inline taipy/core/task/task.py]:
```suggestion
                If not provided and *wait* is True, the function waits indefinitely.
```

**Review by FabienLelaquais — DISMISSED** [review]
👍

**Review by toan-quach — DISMISSED** [review]

**Review by joaoandre-avaiga — APPROVED** [review]

**Pull request merged**